driver-core/src/main/java/com/datastax/driver/core/EventDebouncer.java

```diff
@@ -166,7 +169,7 @@ private void scheduleImmediateDelivery() {
         while (state == State.RUNNING) {
             DeliveryAttempt previous = immediateDelivery.get();
             if (previous != null && !previous.isDone())
-                return;
+                previous.cancel();
```

Comment:
Absolutely, even the poll() method doesn't block. I decided to keep this call because 1) the intention counts as well (we would like the execution to be canceled, even if we know it's not possible) and 2) the callback could theoretically block and/or respond to interrupt requests. But I'm fine if you prefer to remove this instruction completely.